File: backend/notifications/src/main/java/com/oxygensend/notifications/config/NotificationProperties.java
package com.oxygensend.notifications.config;

import jakarta.validation.constraints.NotNull;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

import java.util.Set;

@Validated
@ConfigurationProperties(prefix = "notifications")
public record NotificationProperties(Set<String> services,
                                     String secret,
                                     @NotNull
                                     boolean authEnabled
) {
}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/config/NotificationProperties.kt
package com.oxygensend.notifications.config

import jakarta.validation.constraints.NotNull
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "notifications")
data class NotificationProperties(
    val services: Set<String>,
    val secret: String,
    val authEnabled: @NotNull Boolean
)
